pages/runtime/supervisor-api.md

```diff
@@ -0,0 +1,349 @@
+
+In order to do this, users can create a file called `/data/resin-updates.lock` that will prevent the Supervisor from killing and restarting the app. As any other lockfile, the Supervisor itself will create such file before killing the app, so you should only create it in exclusive mode. This means: only create the lockfile if it doesn't already exist. Several tools exist to simplify this process, for example [npm/lockfile](https://github.com/npm/lockfile).
+
+Using the above-mentioned library, the lock can be acquired like in this CoffeeScript example:
+```coffeescript
+	lockFile = require 'lockfile'
+
+	lockFile.lock '/data/resin-updates.lock', (err) ->
+		# A non-null err probably means the supervisor is about to kill us
+		throw new Error('Could not acquire lock: ', err) if err?
+
+		# Here we have the lock, so we can do critical stuff:
+		doTheHarlemShake()
+
+		# Now we release the lock, and we can be killed again
+		lockFile.unlock '/data/resin-updates.lock', (err) ->
+			# If err is not null here, something went really wrong
+			throw err if err?
+```
+
+There are other libraries you can use in different languages, for example [this Python library](http://pythonhosted.org/lockfile/lockfile.html#examples).
+
+### Overriding the lock
+
+The update lock can be overriden in case you need to force an update, for instance, if your app has hung in a critical section.
+
+The way to do this is hitting the `/v1/update` endpoint of the supervisor HTTP API, with `{ "force": true }` as body. See below for the full API reference. More interfaces to override this lock will be added soon.
+
+## HTTP API reference
+
+The supervisor exposes an HTTP API on port 48484 (`RESIN_SUPERVISOR_PORT`).
+
+**All endpoints require an apikey parameter, which is exposed to the application as `RESIN_SUPERVISOR_API_KEY`.**
+
+The full address for the API, i.e. `"http://127.0.0.1:48484"`, is available as `RESIN_SUPERVISOR_ADDRESS`. **Always use these variables when communicating via the API, since address and port could change**.
+
+Alternatively, the Resin API (api.resin.io) has a proxy endpoint at `POST /supervisor/<url>` from which you can send API commands to the supervisor remotely, using your Auth Token instead of your API key. Commands sent through the proxy require an `appId` and/or `deviceId` parameter in the body, and default to POST requests unless you specify a `method` parameter (e.g. "GET").
+
+The API is versioned (currently at v1), except for `/ping`.
+
+You might notice that the formats of some responses differ. This is because they were implemented later, and in Go instead of node.js.
+
+Here's the list of endpoints implemented so far that are meant to be used by the apps. In all examples, replace everything between `< >` for the corresponding values.
+
+The full reference can be found in the [Supervisor's API docs](https://github.com/resin-io/resin-supervisor/blob/master/docs/API.md).
+
+<hr>
+
+### GET /ping
+
```

Comment:
might be worth having a little explanation of what this ping is for or does. It is the device pinging the resin server API right?